sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ToNumberParser.scala

```diff
@@ -599,4 +612,227 @@ class ToNumberParser(numberFormat: String, errorOnFail: Boolean) extends Seriali
+            }
+          }
+        }
+      case _: DecimalPoint =>
+        reachedDecimalPoint = true
+      case _ =>
+    }
+    // If there were more digits in the provided input string (before or after the decimal point)
+    // than specified in the format string, the input string does not match the format.
+    if (numFormatDigitsBeforeDecimalPoint < numInputDigitsBeforeDecimalPoint ||
+        numFormatDigitsAfterDecimalPoint < numInputDigitsAfterDecimalPoint) {
+      return None
+    }
+    val leadingSpaces = " " * (numFormatDigitsBeforeDecimalPoint - numInputDigitsBeforeDecimalPoint)
+    val trailingSpaces = " " * (numFormatDigitsAfterDecimalPoint - numInputDigitsAfterDecimalPoint)
+    Some((leadingSpaces + beforeDecimalPointBuilder.toString,
+      afterDecimalPointBuilder.toString + trailingSpaces))
+  }
+
+  /**
+   * Performs format processing on the digits in [[groups]], updating [[result]].
+   *
+   * @param groups the token representing a group of digits from the format string
+   * @param inputBeforeDecimalPoint string representation of the input decimal value before the
+   *                                decimal point
+   * @param inputAfterDecimalPoint string representation of the input decimal value after the
+   *                                decimal point
+   * @param reachedDecimalPoint true if we have reached the decimal point so far during processing
+   * @param result the result of formatting is built here as a string during iteration
+   */
+  def formatDigitGroups(
+      groups: DigitGroups,
+      inputBeforeDecimalPoint: String,
+      inputAfterDecimalPoint: String,
+      reachedDecimalPoint: Boolean,
+      result: StringBuilder): Unit = {
+    // Iterate through the tokens in the DigitGroups. Reverse the order of the tokens so we
+    // consume them in the left-to-right order that they originally appeared in the format
+    // string.
+    for (digitGroupToken <- groups.tokens.reverse) {
+      digitGroupToken match {
+        case digits: Digits if !reachedDecimalPoint =>
+          val numDigits = digits match {
+            case ExactlyAsManyDigits(num) => num
+            case AtMostAsManyDigits(num) => num
+          }
+          for (_ <- 0 until numDigits) {
+            inputBeforeDecimalPoint(formattingBeforeDecimalPointIndex) match {
+              case c: Char if c != ' ' =>
+                result.append(c)
+              case _ => digits match {
+                case _: ExactlyAsManyDigits =>
+                  // The format string started with a 0 and had more digits than the provided
+                  // input string, so we prepend a 0 to the result.
+                  result.append('0')
+                case _: AtMostAsManyDigits =>
+                  // The format string started with a 9 and had more digits than the provided
+                  // input string, so we prepend a space to the result.
+                  if (result.nonEmpty && result.last == '<') {
```

Comment:
We should see `>` at the end, not `<` I think

Reply:
The goal is just to check if we just added space after a '>' or '-', and if so, swap the characters. I added a comment to clarify this.